galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/AccessCps.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed Materials - Property of IBM
+ * 
+ * (c) Copyright IBM Corp. 2019.
+ */
+package dev.galasa.framework.api.cps.internal;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+/**
+ * CPS API
+ * 
+ * Allows for CPS properties to be retrieved and added
+ * 
+ */
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
+    "osgi.http.whiteboard.servlet.pattern=/cps/*" }, name = "Galasa CPS")
+public class AccessCps extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+
+    private Log logger = LogFactory.getLog(getClass());
+
+    private final Gson gson = GalasaGsonBuilder.build();
+
+    @Reference
+    public IFramework framework; // NOSONAR
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String[] reqParams = req.getPathInfo().substring(1).split("/");
+        if(reqParams.length == 1 && reqParams[0].equals("namespace")) {
+            JsonArray namespaceArray = new JsonArray();
+            try {
+                List<String> namespaces = framework.getConfigurationPropertyService("framework").getCPSNamespaces();
+                for(String name : namespaces) {
+                    namespaceArray.add(name);
+                }
+            } catch(ConfigurationPropertyStoreException e) {
+                logger.error("Unable to access CPS", e);
+                resp.setStatus(500);
```

Comment:
you need to give the caller a explaination of the failure without having to go to the microservice, because there may be 10s of servers.
Return an error json object?

Reply:
will need to update the openapi format, need to return the stacktrace